classes/fields/avatar.php

```diff
@@ -79,39 +79,161 @@ public function input( $name, $value = null, $options = null, $pod = null, $id =
-			$user_id = (int) $id_or_email->ID;
-		} elseif ( ! is_object( $id_or_email ) && false !== strpos( $id_or_email, '@' ) ) {
-			$_user = get_user_by( 'email', $id_or_email );
+	/**
+	 * Take over the avatar data served from WordPress
+	 *
+	 * @since 2.7.22
+	 *
+	 * @param array $args {
+	 *     Optional. Arguments to return instead of the default arguments.
+	 *
+	 *     @type int    $size           Height and width of the avatar image file in pixels. Default 96.
+	 *     @type int    $height         Display height of the avatar in pixels. Defaults to $size.
+	 *     @type int    $width          Display width of the avatar in pixels. Defaults to $size.
+	 *     @type string $default        URL for the default image or a default type. Accepts '404' (return
+	 *                                  a 404 instead of a default image), 'retro' (8bit), 'monsterid' (monster),
+	 *                                  'wavatar' (cartoon face), 'indenticon' (the "quilt"), 'mystery', 'mm',
+	 *                                  or 'mysteryman' (The Oyster Man), 'blank' (transparent GIF), or
+	 *                                  'gravatar_default' (the Gravatar logo). Default is the value of the
+	 *                                  'avatar_default' option, with a fallback of 'mystery'.
+	 *     @type bool   $force_default  Whether to always show the default image, never the Gravatar. Default false.
+	 *     @type string $rating         What rating to display avatars up to. Accepts 'G', 'PG', 'R', 'X', and are
+	 *                                  judged in that order. Default is the value of the 'avatar_rating' option.
+	 *     @type string $scheme         URL scheme to use. See set_url_scheme() for accepted values.
+	 *                                  Default null.
+	 *     @type array  $processed_args When the function returns, the value will be the processed/sanitized $args
+	 *                                  plus a "found_avatar" guess. Pass as a reference. Default null.
+	 *     @type string $extra_attr     HTML attributes to insert in the IMG element. Is not sanitized. Default empty.
+	 * }
+	 * @param mixed $id_or_email The Gravatar to retrieve. Accepts a user ID, Gravatar MD5 hash,
+	 *                           user email, WP_User object, WP_Post object, or WP_Comment object.
+	 * @return array {
+	 *     Along with the arguments passed in `$args`, this will contain a couple of extra arguments.
+	 *
+	 *     @type bool   $found_avatar True if we were able to find an avatar for this user,
+	 *                                false or not set if we couldn't.
+	 *     @type string $url          The URL of the avatar we found.
+	 * }
+	 */
+	public function get_avatar_data( $args, $id_or_email ) {
+		if ( ! $this->allow_avatar_overwrite() ) {
+			return $args;
+		}
 
-			if ( ! empty( $_user ) ) {
-				$user_id = (int) $_user->ID;
+		$return_args = $args;
+
+		$args = wp_parse_args(
+			$args,
+			array(
```

Comment:
:no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: be84cfc77ca9c6a3452e2ab603bb59b0</sub>